test/contains.js

```diff
@@ -28,6 +28,64 @@ describe('contains', function() {
     assert.strictEqual(R.contains(new Just([42]), [new Just([42])]), true);
   });
 
+  it('dispatches to `contains` method', function() {
+    function Empty() {}
+    Empty.prototype.contains = R.always(false);
+
+    function List(head, tail) {
+      this.head = head;
+      this.tail = tail;
+    }
+    List.prototype.contains = function(x) {
+      return this.head === x || this.tail.contains(x);
+    };
+
+    var list = new List('foo', new List('bar', new List('baz', new Empty())));
+
+    assert.strictEqual(R.contains('foo', list), true);
+    assert.strictEqual(R.contains('bar', list), true);
+    assert.strictEqual(R.contains('baz', list), true);
+    assert.strictEqual(R.contains('quux', list), false);
+  });
+
+  it('dispatches to `indexOf` method', function() {
```

Comment:
This test does not belong. this locks in the implementation of `contains` to forever be dependent on `indexOf`. The definition of `contains` is "returns true if the element is inside the container," _not_ "returns true if the index of the element in the container is greater than zero."